src/logger-helpers.ts

```diff
@@ -26,32 +26,38 @@ export const isLoggerInstance = (subject: unknown): subject is AbstractLogger =>
   isObject(subject) &&
   Object.keys(severity).some((method) => method in subject);
 
-/**
- * @todo consider Intl units when Node 18 dropped (microsecond unit is missing, picosecond is not in list)
- * @link https://tc39.es/ecma402/#table-sanctioned-single-unit-identifiers
- * */
-const makeNumberFormat = (fraction = 0) =>
+/** @link https://tc39.es/ecma402/#table-sanctioned-single-unit-identifiers */
+type TimeUnit =
+  | "nanosecond"
+  | "microsecond"
+  | "millisecond"
+  | "second"
+  | "minute";
+
+const makeNumberFormat = (unit: TimeUnit, fraction = 0) =>
   Intl.NumberFormat(undefined, {
     useGrouping: false,
     minimumFractionDigits: 0,
     maximumFractionDigits: fraction,
+    style: "unit",
+    unitDisplay: "long",
+    unit,
   });
 
 // creating them once increases the performance significantly
-const intFormat = makeNumberFormat();
-const floatFormat = makeNumberFormat(2);
+const picoFormat = makeNumberFormat("nanosecond", 3);
+const nanoFormat = makeNumberFormat("nanosecond");
+const microFormat = makeNumberFormat("microsecond");
+const milliFormat = makeNumberFormat("millisecond");
+const secondsFormat = makeNumberFormat("second", 2);
+const minutesFormat = makeNumberFormat("minute", 2);
```

Comment:
should move it into class props, so that they're not allocated when using custom logger

Reply:
will address in another PR

Reply:
addressed